tezedge_state/src/state/assert_state.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::collections::BTreeMap;
+use std::fmt::{self, Display};
+
+use crate::TezedgeState;
+
+#[derive(Debug, Clone, Copy)]
+enum PeerContainer {
+    Potential,
+    Pending,
+    Connected,
+    Blacklisted,
+}
+
+impl fmt::Display for PeerContainer {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::Potential => "potential_peers",
+                Self::Pending => "pending_peers",
+                Self::Connected => "connected_peers",
+                Self::Blacklisted => "blacklisted_peers",
+            }
+        )
+    }
+}
+
+#[inline]
+fn assert_peer_counts(state: &TezedgeState) {
+    assert!(state.potential_peers.len() <= state.config.max_potential_peers);
+    assert!(state.pending_peers.len() <= state.config.max_pending_peers);
+    assert!(state.connected_peers.len() <= state.config.max_connected_peers);
+}
+
+#[inline]
+fn assert_no_overlapping_peers(state: &TezedgeState) {
+    fn panic_overlap<A: Display>(addr: A, first: PeerContainer, second: PeerContainer) {
+        panic!(
```

Comment:
@binier @jurajselep 
I am really not sure about this panic and asserts,
do we really want to crash the node in production with them?

shouldnt it be more `debug_assert*` just in debug mode?

Reply:
Those aren't used in running node, just testing/fuzzing. Maybe that needs to be under cfg flag as well,

Reply:
Also `debug_assert` is not enough, because fuzzer compiles node in `release` mode as well as when doing simulations or other tests.